provisioner/puppet-masterless/provisioner.go

```diff
@@ -227,8 +275,15 @@ func (p *Provisioner) Provision(ui packer.Ui, comm packer.Communicator) error {
 	}
 
 	// Execute Puppet
+	var facterVarsString string
+	if p.config.GuestOSType == provisioner.UnixOSType {
+		facterVarsString = strings.Join(facterVars, " ")
+	} else {
+		facterVarsString = "SET " + strings.Join(facterVars, " && SET ")
+	}
+
 	p.config.ctx.Data = &ExecuteTemplate{
```

Comment:
Ok I had a look, puppet-server uses `p.guestOSTypeConfig.facterVarsFmt` and puppet-master should probably do something similar.